mempool/mempool.go

```diff
@@ -296,6 +298,10 @@ func (mem *Mempool) TxsWaitChan() <-chan struct{} {
 //     It gets called from another goroutine.
 // CONTRACT: Either cb will get called, or err returned.
 func (mem *Mempool) CheckTx(tx types.Tx, cb func(*abci.Response)) (err error) {
+	return mem.checkTxFromPeer(tx, cb, 0)
+}
+
+func (mem *Mempool) checkTxFromPeer(tx types.Tx, cb func(*abci.Response), peerID uint16) (err error) {
```

Comment:
I think this method should be public. Like CheckTx it's a part of the public interface for all the consumers (reactor or anybody else)

Reply:
Good point. I do find it weird to have both CheckTx and CheckTxFromPeer in a public interface. Should we then just have "CheckTxFromPeer" and have UnknownPeerID be exported as well?

Reply:
> Should we then just have "CheckTxFromPeer" and have UnknownPeerID be exported as well?

Yes!

Reply:
Not having CheckTx at all looked a bit odd to me, do you still think it should be removed?